swap_meet/vendor.py

```diff
@@ -1,2 +1,229 @@
+
+"""
+
+from asyncio.format_helpers import _format_callback_source
+
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    #with .pop()
+    def remove(self, item):
+        for i in range(len(self.inventory)):
+            if self.inventory[i] == item:
+                removed = self.inventory.pop(i)
+                return removed
+        return False
+
+    #with .remove()
+    # def remove(self,item):
+    #     aList = self.inventory
+    #     for thing in aList:
+    #         if thing == item:
+    #             self.inventory.remove(thing)
+    #             return thing
+    #     return False
+
+    def get_by_category(self, category):
+        itemList = []
+        
+        for item in self.inventory:
+            if item.category == category:
+                itemList.append(item)
+        #returns a list of `Item`s in the inventory with that category
+        return itemList
+
+    # using list comprehension, don't have to use append
+    # itemList = [item for item in self.inventory if item.category == category]
+    """
+    
+    To further reduce the amount of repeated code in your project, 
+    consider how `swap_best_by_category` and `swap_first_item` might be able to make use of `swap_items`. 
+    Is there a way that these methods could incorporate a call to `swap_items` into the body of these methods?
+
+    ###To do this, could create default paremeters or is it keyword arguments for my_item and their item in swap items
+    then create an if, elif situation in swap items, then replace the expression in the if statement of swap_first_item with
+    the swap_items function ???? not sure what the default parameters/keyword args would be tho...
+    update: Oh my gosh I don't know why I thought about this like this, can just use in the other funcs like Hailey and Philomena did....
+    
+    """    
+    
+    def swap_items(self,friend, my_item, their_item):
+        #If this `Vendor`'s inventory doesn't contain `my_item` or the friend's inventory doesn't contain `their_item`, the method returns `False`
+        if my_item not in self.inventory or their_item not in friend.inventory:
+            return False
+
+        #remove the `my_item` from this `Vendor`'s inventory, and add it to the friend's inventory
+        my_removed = self.remove(my_item)
+        friend.inventory.append(my_removed)
+
+        #remove the `their_item` from the other `Vendor`'s inventory, and add it to this `Vendor`'s inventory
+        their_removed = friend.remove(their_item)
+        self.inventory.append(their_removed)
+        
+        return True
```

Comment:
👍  Nice work using the return values from the Vendor's `remove` and `add` methods.